Assignment 1/src/main/java/com/cezaram28/Assignment1/controller/ConsoleController.java

```diff
@@ -0,0 +1,676 @@
+                handleAddQuestion();
+                return false;
+            case "removeQuestion":
+                handleRemoveQuestion();
+                return false;
+            case "editQuestion":
+                handleEditQuestion();
+                return false;
+            case "listAnswers":
+                handleListAnswers();
+                return false;
+            case "listAnswersByQuestion":
+                handleListAnswersQuestion();
+                return false;
+            case "listAnswersByUser":
+                handleListAnswersUser();
+                return false;
+            case "addAnswer":
+                handleAddAnswer();
+                return false;
+            case "removeAnswer":
+                handleRemoveAnswer();
+                return false;
+            case "editAnswer":
+                handleEditAnswer();
+                return false;
+            case "searchTitle":
+                handleSearchTitle();
+                return false;
+            case "searchTag":
+                handleSearchTag();
+                return false;
+            case "upvoteQuestion":
+                handleUpvoteQuestion();
+                return false;
+            case "downvoteQuestion":
+                handleDownvoteQuestion();
+                return false;
+            case "upvoteAnswer":
+                handleUpvoteAnswer();
+                return false;
+            case "downvoteAnswer":
+                handleDownvoteAnswer();
+                return false;
+            case "makeAdmin":
+                makeAdmin();
+                return false;
+            case "ban":
+                ban();
+                return false;
+            case "exit":
+                return true;
+            default:
+                print("Unknown command. Try again.");
+                return false;
+        }
+    }
+
+    private void login() {
+        print("Username:");
+        String username = scanner.next().trim();
+        print("Password:");
+        String password = scanner.next().trim();
+        Optional<User> user = userManagementService.findByCredentials(username,password);
+        if(user.isPresent()){
+            if(user.get().getIsBanned())
+                print("You are banned indefinitely");
+            else
+                this.user = user.get();
+        } else print("Bad credentials!");
+    }
+
+    private void logout(){
+        user = null;
+        print("Logged out");
+    }
+
+    private void handleListUsers() {
+        if(user==null) print("Please log in");
+        else
+        userManagementService.listUsers().forEach(u->print(u.toString()));
+    }
+
+    private void handleListQuestions() {
+        if(user==null) print("Please log in");
+        else {
+            List<Question> questions = questionManagementService.listQuestions();
+            questions.sort((q1,q2)->q1.getCreationDate().after(q2.getCreationDate())?-1:1);
```

Comment:
You can simply do q1.getCreationDate().compareTo(q2.getCreationDate()) instead (also applicable for the rest of the cases in which you do sort).